src/cml.js

```diff
@@ -591,14 +591,18 @@ class CML {
       if (url) return renderPr(url);
     } else {
       await exec('git', 'fetch', remote, sha);
-      await exec('git', 'checkout', '-B', target, sha);
       await exec('git', 'checkout', '-b', source);
-      await exec('git', 'add', paths.join(' '));
-      let commitMessage = message || `CML PR for ${shaShort}`;
-      if (skipCi || (!message && !(merge || rebase || squash))) {
-        commitMessage += ' [skip ci]';
+      await exec('git', 'checkout', '-B', target, sha);
+
+      if (paths.length) {
+        await exec('git', 'add', paths.join(' '));
+        let commitMessage = message || `CML PR for ${shaShort}`;
+        if (skipCi || (!message && !(merge || rebase || squash))) {
```

Comment:
This begs a comment explaining why.

Reply:
(I understand that this line was not introduced in this PR)

Reply:
yes, that is ugly and partially my fault.